src/ray/common/ray_config_def.h

```diff
@@ -92,6 +92,12 @@ RAY_CONFIG(size_t, free_objects_batch_size, 100)
 
 RAY_CONFIG(bool, lineage_pinning_enabled, false)
 
+/// Maximum amount of lineage to keep. This includes the specs of all tasks
+/// that have previously already finished but that may be retried again. If we
+/// reach this limit, all current lineage will be evicted and objects that are
+/// still in scope will no longer be reconstructed if lost.
+RAY_CONFIG(int64_t, max_lineage_bytes, 1024 * 1024 * 1024)
```

Comment:
Just curious, but how big is each task spec is normally? Maybe we can mention how many of lineages can be approximately cached?

Reply:
Good point, I'll add this. Task spec with no args is about 500 bytes, but it could go up a lot if there are a lot of inlined args.